python/ray/data/_internal/plan.py

```diff
@@ -415,6 +435,23 @@ def can_fuse(self, prev: Stage):
         # not the other way around. The latter will be used as the compute if fused.
         if is_task_compute(self.compute) and prev.compute != self.compute:
             return False
+        if prev.fn_constructor_args != self.fn_constructor_args:
+            return False
+        if prev.fn_constructor_kwargs != self.fn_constructor_kwargs:
+            return False
```

Comment:
What's the reason they have to be the same? If the prev is a task and self is CallableClass/actor, or if they are both CallableClasses but run different fn (or same fn with different construction args), these seem fine?

Reply:
For the multiple callable classes case (difference callable classes, or same callable class with different construction args), we can't support this since we can't initialize multiple callable classes in the actor constructor. Callable classes are initialized once, in the actor constructor when an actor worker is created, and we can only initialize one callable class at a time since we need to provide `fn_constructor_args` and `fn_constructor_kwargs` as flat arguments so object references get properly resolved in the actor creation task.

For the normal function + callable class case, this is actually supposed to be supported (`fuse()` is designed to support it), so I think that this fuse check is too strict. Let me fix that!

Reply:
It'd be great to comment the first case for the line that performs the check, and add a test case for them.